app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsContributionsFragment.kt

```diff
@@ -28,32 +30,27 @@ import org.wikipedia.suggestededits.Contribution.Companion.EDIT_TYPE_ARTICLE_DES
 import org.wikipedia.suggestededits.Contribution.Companion.EDIT_TYPE_IMAGE_CAPTION
 import org.wikipedia.suggestededits.Contribution.Companion.EDIT_TYPE_IMAGE_TAG
 import org.wikipedia.util.DateUtil
+import org.wikipedia.util.DimenUtil
 import org.wikipedia.util.ResourceUtil
 import org.wikipedia.util.log.L
 import org.wikipedia.views.DefaultViewHolder
 import java.util.*
+import java.util.concurrent.TimeUnit
 import kotlin.collections.ArrayList
 import kotlin.collections.HashMap
 
-class SuggestedEditsContributionsFragment : Fragment(), SuggestedEditsTypeItem.Callback {
+class SuggestedEditsContributionsFragment : Fragment(), SuggestedEditsContributionsHeaderView.Callback {
     private val adapter: ContributionsEntryItemAdapter = ContributionsEntryItemAdapter()
-    private var articleContributions = ArrayList<Contribution>()
-    private var imageContributions = HashSet<Contribution>()
-    private var articleAndImageContributions = ArrayList<Contribution>()
-    private var consolidatedContributionsWithDates: MutableList<Any> = ArrayList()
-    private var continuedArticlesContributions = ArrayList<Contribution>()
-    private val continuedImageContributions = HashSet<Contribution>()
+
```

Comment:
@dbrant Removing the hashsets has resulted in multiple entries one  for each image tag added... also resulting in multiple network calls to image info..

Reply:
Don't we *want* multiple entries, one for each image tag?  Attempting to "coalesce" multiple edits into one item will not work well, and is bound to explode in complexity.  We're building a list of edits, not a list of articles that you have edited.

Reply:
I agree with that school of thought, but clarified during review and during discussions with Robin that these had to be one entry ...yes.. that did take me some time and complexity to coalesce and group the tags while coding this. Might need to check in with Charlotte and Robin